### ISSUE TITLE
Crash during sleep (on wakeup?)

### ISSUE THREAD

**Issue by RenechCDDA** [CleverRaven/Cataclysm-DDA#69703]
### Describe the bug

It crashes. I have absolutely no clue.

```The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 02f75a2
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

  #0
    (dbghelp: debug_write_backtrace+0xb5@0,000,7FF,6AF,FB7,295[cataclysm-tiles.exe+0x557,295]), 
  #1
    (dbghelp: log_crash+0x377@0,000,7FF,6AF,F79,4B7[cataclysm-tiles.exe+0x519,4b7]), 
  #2
    (dbghelp: signal_handler+0x57@0,000,7FF,6AF,F79,A47[cataclysm-tiles.exe+0x519,a47]), 
  #3
    (dbghelp: seh_filter_exe+0x178@0,000,7FF,6B1,4ED,5D8[cataclysm-tiles.exe+0x1,a8d,5d8]), 
  #4
    (dbghelp: `__scrt_common_main_seh'::`1'::filt$0+0x16@0,000,7FF,6B1,722,416[cataclysm-tiles.exe+0x1,cc2,416]), 
  #5
    (dbghelp: _C_specific_handler+0xa0@0,000,7FF,6B1,4BE,814[cataclysm-tiles.exe+0x1,a5e,814]), 
  #6
    (dbghelp: _chkstk+0x11f@0,000,7FF,FE6,292,3AF[ntdll.dll+0xa2,3af]), 
  #7
    (dbghelp: RtlRaiseException+0x484@0,000,7FF,FE6,241,4B4[ntdll.dll+0x51,4b4]), 
  #8
    (dbghelp: KiUserExceptionDispatcher+0x2e@0,000,7FF,FE6,290,EBE[ntdll.dll+0xa0,ebe]), 
  #9
    (dbghelp: effect_type::lose_game_message_type+0x3b@0,000,7FF,6B0,8C5,E9B[cataclysm-tiles.exe+0xe65,e9b]),
```

### Attach save file

[Pawnee Station-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13460495/Pawnee.Station-trimmed.tar.gz)

### Steps to reproduce

Load save, immediately try to sleep ($).

Repeatedly tell it "keep trying to fall asleep" (not "and do not ask me again")

Does not reproduce 100% of the time

### Expected behavior

Peaceful sleep only interrupted by portal storms and my companions wandering around making noises

### Versions and configuration

- OS: Windows
    - OS Version: 10
- Game Version: 02f75a2 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Translate Complex Dialogue [translate_dialogue]
]

### Additional context

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/13460512/crash.log)

**Comment by akrieger** [comment]
Full stack.
```
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 0410d8d
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

  #0
    (dbghelp: debug_write_backtrace+0xb5@0,000,7FF,64C,907,295[cataclysm-tiles.exe+0x557,295]), 
  #1
    (dbghelp: log_crash+0x377@0,000,7FF,64C,8C9,4B7[cataclysm-tiles.exe+0x519,4b7]), 
  #2
    (dbghelp: signal_handler+0x57@0,000,7FF,64C,8C9,A47[cataclysm-tiles.exe+0x519,a47]), 
  #3
    (dbghelp: seh_filter_exe+0x178@0,000,7FF,64D,E3D,618[cataclysm-tiles.exe+0x1,a8d,618]), 
  #4
    (dbghelp: `__scrt_common_main_seh'::`1'::filt$0+0x16@0,000,7FF,64E,072,3A6[cataclysm-tiles.exe+0x1,cc2,3a6]), 
  #5
    (dbghelp: _C_specific_handler+0xa0@0,000,7FF,64D,E0E,854[cataclysm-tiles.exe+0x1,a5e,854]), 
  #6
    (dbghelp: _chkstk+0x12f@0,000,7FF,905,654,41F[ntdll.dll+0xa4,41f]), 
  #7
    (dbghelp: RtlFindCharInUnicodeString+0xa96@0,000,7FF,905,5CE,466[ntdll.dll+0x1e,466]), 
  #8
    (dbghelp: KiUserExceptionDispatcher+0x2e@0,000,7FF,905,653,40E[ntdll.dll+0xa3,40e]), 
  #9
    (dbghelp: effect_type::lose_game_message_type+0x3b@0,000,7FF,64D,215,E8B[cataclysm-tiles.exe+0xe65,e8b]), 
  #10
    (dbghelp: Creature::remove_effect+0x193@0,000,7FF,64C,BC2,803[cataclysm-tiles.exe+0x812,803]), 
  #11
    (dbghelp: Creature::process_effects+0x745@0,000,7FF,64C,BC2,425[cataclysm-tiles.exe+0x812,425]), 
  #12
    (dbghelp: Creature::process_turn+0x4d@0,000,7FF,64C,BC2,55D[cataclysm-tiles.exe+0x812,55d]), 
  #13
    (dbghelp: Character::process_turn+0x2bf@0,000,7FF,64C,C25,C8F[cataclysm-tiles.exe+0x875,c8f]), 
  #14
    (dbghelp: do_turn+0x1,281@0,000,7FF,64C,90E,FF1[cataclysm-tiles.exe+0x55e,ff1]), 
  #15
    (dbghelp: WinMain+0xc85@0,000,7FF,64C,794,BA5[cataclysm-tiles.exe+0x3e4,ba5]), 
  #16
    (dbghelp: __scrt_common_main_seh+0x106@0,000,7FF,64D,E0B,48A[cataclysm-tiles.exe+0x1,a5b,48a]), 
  #17
    (dbghelp: BaseThreadInitThunk+0x1d@0,000,7FF,903,FE2,57D[KERNEL32.DLL+0x12,57d]), 
  #18
    (dbghelp: RtlUserThreadStart+0x28@0,000,7FF,905,60A,A58[ntdll.dll+0x5a,a58]), 
```

**Comment by andrei8l** [comment]
<details>
<summary>Always reproducible with _GLIBCXX_ASSERTIONS</summary>

```gdb
/usr/include/c++/13.2.1/bits/stl_vector.h:1144: std::vector<_Tp, _Alloc>::const_reference std::vector<_Tp, _Alloc>::operator[](size_type) const [with _Tp = std::pair<translation, game_message_type>; _Alloc = std::allocator<std::pair<translation, game_message_type> >; const_reference = const std::pair<translation, game_message_type>&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

</details>

This `<` should probably be a `>`
https://github.com/CleverRaven/Cataclysm-DDA/blob/5ec89597b0cb617ac44c8c4608be60045c56de4b/src/effect.cpp#L554-L556

**Comment by RenechCDDA** [comment]
Should be resolved by #70420
